### ISSUE TITLE
(macOS) Set output device without setting system default and/or set null output

### ISSUE THREAD

**Issue by thislooksfun** [audiokit/AudioKit#632]
There's two (related) parts to this request:
1. I'm attempting to make a microphone FFT viewer, and thanks to #546, I can now select the input device I want to use without setting the system default value, but since I just want to view the plot, I don't want to play the audio out of the output. I tried the following, but I was still able to hear the output in my headphones:
```swift
let mic = AKMicrophone()
do {
	try mic.setDevice(AudioKit.availableInputs![1])
} catch {}
let mixer = AKMixer(mic)

let out = AKMixer(mixer)
out.volume = 0

AudioKit.output = out
AudioKit.start()

out.start()

let plot = AKNodeFFTPlot(mixer, frame: CGRect(x: 0, y: 0, width: 500, height: 500))
```

2. The other request/option is to be able to select the output device used without setting the system default. This way I can pipe audio through headphones or speakers as I wish, and I (and other developers) can easily route audio between apps, like with Soundflower.

Thanks,
-tlf

P.S. Fantastic job so far, and thanks for being so helpful. Keep up the great work!

**Comment by aure** [comment]
I'm glad you found these bugs, definitely did not make sense the old way.

**Comment by thislooksfun** [comment]
Thanks for the prompt update, but does this allow me to set the output device to null to disable playback?

**Comment by aure** [comment]
Well, AudioKit needs an output that is requesting the rest of the nodes to provide data.  So that null would have to do something.  What would you use it for?  Can you just put a booster at the end of the chain bringing down the volume to zero?

**Comment by thislooksfun** [comment]
Well that was part one of this issue. I'm putting the microphone through an `AKMixer` to fix the problem of #631, and then putting the output of that through another `AKMixer` on which I'm setting the volume to 0, but I can still hear the output. I will try it again soon (I'm taking a very short break from a paper right now), but I'm not sure why that was happening.

**Comment by aure** [comment]
Instead of out = AKMixer(mixer), try out = AKBooster(mixer, gain: 0)
